types/rpc-handler.ts

```diff
@@ -71,7 +71,7 @@ export class RPCHandler implements HandlerInterface {
       throw new Error("Failed to find fastest RPC");
     }
 
-    const provider = new JsonRpcProvider(fastestRpcUrl, this._networkId);
+    const provider = new JsonRpcProvider(fastestRpcUrl, Number(this._networkId));
```

Comment:
Seems strange to me that you defined them as strings and then coerce to numbers here.

Reply:
I hear you, it's because of the reasons above regarding the types but I can revert things back to using numbers I'd just need to update and types and consts but the types won't be a clean/simple